### DISCUSSION TITLE
Help with debugging when Brighscript exits unexpectedly

### DISCUSSION THREAD

**Discussion by rrirower** [rokucommunity/vscode-brightscript-language#573] [Q&A]
My Roku channel app is crashing the device and rebooting.  I want to be able to identify the cause, but, I can't seem to trap the error.  The device hangs, then reboots if I try to play a Live stream (ie. m3u8).  I get this message in the console when the crash occurs:

_App exit event detected and launchConfiguration.stopDebuggerOnAppExit is true - shutting down debug session_

I've tried disabling the OnAppExit switch but it does not help.  I've also tried wrapping Try/Catch around code I suspect.

Is there a recommended way to trap this kind of error?

**Comment by TwitchBronBron** [comment]
Are you using the debug protocol? If so, try using telnet instead. (`"enableDebugProtocol": false`). Then also set ``stopDebuggerOnAppExit": false` in your `launch.json` and see if anything more useful shows up in the output logs.

**Reply by rrirower** [reply]
I had tried using telnet with those settings.  However, this is a pretty hard crash and it closed my telnet client.  I did have a telnet log enabled, but, it only shows warnings.

**Reply by TwitchBronBron** [reply]
Most of the time these types of crashes are caused by the Roku firmware itself. The debug protocol and telnet debug sessions still have random crashes completely outside of our (the vscode extension's) control. 

You can try enabling verbose logging by enabling verbose logging and see if there's anything useful in there, but most likely there's nothing we can do since the device just hung up on us.

Add these to your `launch.json`
```json
            "logLevel": "debug",
            "fileLogging":{
                "dir": "${workspaceFolder}/logs",
                "rokuDevice": {
                    "enabled": true,
                    "filename": "rokuDevice.log",
                    "mode": "session",
                    "logLimit": 5
                },
                "debugger": {
                    "enabled": true,
                    "filename": "debugger.log",
                    "mode": "session",
                    "logLimit": 5
                }
            }
```
